packages/arb-token-bridge-ui/src/hooks/useOftTransactionHistory.ts

```diff
@@ -0,0 +1,355 @@
+ * eg: https://scan.layerzero-api.com/v1/messages/wallet/0x0000000000000000000000000000000000000000
+ */
+interface LayerZeroMessage {
+  pathway: {
+    srcEid: number
+    dstEid: number
+    sender: {
+      address: string
+      id: string
+      name: string
+      chain: string
+    }
+    receiver: {
+      address: string
+      chain: string
+    }
+    id: string
+    nonce: number
+  }
+  source: {
+    status: string
+    tx: {
+      txHash: string
+      blockHash: string
+      blockNumber: string
+      blockTimestamp: number
+      from: string
+      payload: string
+      readinessTimestamp: number
+      options: {
+        lzReceive: {
+          gas: string
+          value: string
+        }
+        ordered: boolean
+      }
+    }
+  }
+  destination: {
+    nativeDrop: {
+      status: string
+    }
+    lzCompose: {
+      status: string
+    }
+    tx: {
+      txHash: string
+      blockHash: string
+      blockNumber: number
+      blockTimestamp: number
+    }
+    status: string
+  }
+  verification: {
+    dvn: {
+      dvns: {
+        [key: string]: {
+          txHash: string
+          blockHash: string
+          blockNumber: number
+          blockTimestamp: number
+          proof: {
+            packetHeader: string
+            payloadHash: string
+          }
+          optional: boolean
+          status: string
+        }
+      }
+      status: string
+    }
+    sealer: {
+      tx: {
+        txHash: string
+        blockHash: string
+        blockNumber: number
+        blockTimestamp: number
+      }
+      status: string
+    }
+  }
+  guid: string
+  config: {
+    error: boolean
+    receiveLibrary: string
+    sendLibrary: string
+    inboundConfig: {
+      confirmations: number
+      requiredDVNCount: number
+      optionalDVNCount: number
+      optionalDVNThreshold: number
+      requiredDVNs: string[]
+      requiredDVNNames: string[]
+      optionalDVNs: string[]
+      optionalDVNNames: string[]
+    }
+    outboundConfig: {
+      confirmations: number
+      requiredDVNCount: number
+      optionalDVNCount: number
+      optionalDVNThreshold: number
+      requiredDVNs: string[]
+      requiredDVNNames: string[]
+      optionalDVNs: string[]
+      optionalDVNNames: string[]
+      executor: string
+    }
+    ulnSendVersion: string
+    ulnReceiveVersion: string
+  }
+  status: {
+    name: string
+    message: string
+  }
+  created: string
+  updated: string
+}
+
+const getOftTransactionStatus = (message: LayerZeroMessage) => {
+  return (
+    LayerZeroMessageStatus[
+      message.status.name as keyof typeof LayerZeroMessageStatus
+    ] ?? 'pending'
+  )
+}
+
+function validateSourceAndDestinationChainIds(message: LayerZeroMessage) {
+  const sourceChainId = getChainIdFromEid(message.pathway.srcEid)
+  const destinationChainId = getChainIdFromEid(message.pathway.dstEid)
+
+  return sourceChainId && destinationChainId
+}
+
+function mapLayerZeroMessageToMergedTransaction(
```

Comment:
This is more like `mapLayerZeroMessageToLayerZeroTransaction`. `LayerZeroTransaction` -> `MergedTransaction` is done in `transformTransaction`.